pyuvdata/uvdata/mir.py

```diff
@@ -369,43 +566,11 @@ def read_mir(
         self.blt_order = ("time", "baseline")
 
         # set filename attribute
-        basename = filepath.rstrip("/")
+        basename = mir_data.filepath.rstrip("/")
         self.filename = [os.path.basename(basename)]
         self._filename.form = (1,)
 
-        # TODO: Spw axis to be collapsed in future release
-        if dsb_spws:
-            # Gotta do a little bit of sleight-of-hand here, on account of
-            # the fact that the ordering of the sidebands makes absolutely
-            # no sense.
-            data_array = np.concatenate(
-                (
-                    np.reshape(
-                        np.concatenate(
-                            list(compress(mir_data.vis_data, sb_screen[sp_bl_maparr])),
-                        ),
-                        (self.Nblts, 1, self.Nfreqs // 2, self.Npols),
-                    ),
-                    np.reshape(
-                        np.concatenate(
-                            list(compress(mir_data.vis_data, ~sb_screen[sp_bl_maparr])),
-                        ),
-                        (self.Nblts, 1, self.Nfreqs // 2, self.Npols),
-                    ),
-                ),
-                axis=2,
-            )
-        else:
-            # If single sideband, then this is a pretty simple operation
-            data_array = np.reshape(
-                np.concatenate(mir_data.vis_data),
-                (self.Nblts, 1, self.Nfreqs, self.Npols),
-            )
-
-        # Don't need the data anymore, so drop it
-        mir_data.unload_data()
-
-        self.data_array = data_array
+        self.data_array = np.transpose(vis_data, (0, 2, 1))[:, np.newaxis, :, :]
```

Comment:
Which axes need to be transposed here?

Reply:
I added some comments in 5c36be1a, but the most efficient way to read in the data here has the data organized by (blt, pol, freq), whereas `UVData` expects (blt, 1, freq, pol), hence the flip.